### PR TITLE
FLUID-5804: Prefs framework tutorial on persistence; SettingsStore API docs

### PR DESCRIPTION
**Pull request by acheetham**

### CODE REVIEW

File: src/documents/SettingsStore.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+The Infusion [Preferences Framework](PreferencesFramework.md) uses a _Settings Store_ to read and
+write preferences.
+
+## fluid.prefs.store ##
+
+A base grade to be used by any preference setting store.
+This component uses the following base [grades](ComponentGrades.md):
+
+* [`fluid.prefs.dataSource`](#fluid-prefs-datasource)
```
Comment (cindyli) [inline src/documents/SettingsStore.md]: It seems the link for this is "#fluidprefsdatasource" instead of "#fluid-prefs-datasource".

**Comment by cindyli** [comment]
In this doc - UserInterfaceOptionsAPI.md, line 365, the link for "Cookie Settings Store" should be changed to point to the new doc.

**Comment by cindyli** [comment]
What do you think to add a link for the use of "settingsStore" in ConnectingThePartsOfAPreferencesEditor.md, line 13?

File: src/documents/SettingsStore.md (hunk 1 above)
Comment (acheetham) [inline src/documents/SettingsStore.md]: @cindyli, I checked the generated HTML for this, and tested it in three browsers: It works for me as-is.

Comment (acheetham) [inline src/documents/SettingsStore.md]: Filed separately: https://issues.fluidproject.org/browse/FLUID-5850

File: src/documents/Builder.md [hunk 2]
```diff
@@ -144,12 +144,12 @@ The builder object returned by a call to `fluid.prefs.builder()` has the followi
...
 </tr>
 <tr>
     <td><code>options.assembledUIEGrade</code></td>
     <td>String</td>
-    <td><a href="ComponentGrades.md">Grade name</a> of the constructed <a href="to-do/UIEnhancer.md">enhancer</a>; Can be used to instantiate the enhancer and <a href="to-do/SettingsStore.md">settings store</a> using <code><a href="https://github.com/fluid-project/infusion/blob/infusion-1.5/src/framework/core/js/Fluid.js#L930-L944">fluid.invokeGlobalFunction</a></code>.</td>
+    <td><a href="ComponentGrades.md">Grade name</a> of the constructed <a href="to-do/UIEnhancer.md">enhancer</a>; Can be used to instantiate the enhancer and <a href="SettingsStore.md">settings store</a> using <code><a href="https://github.com/fluid-project/infusion/blob/infusion-1.5/src/framework/core/js/Fluid.js#L930-L944">fluid.invokeGlobalFunction</a></code>.</td>
```
Comment (jobara) [inline src/documents/Builder.md]: This table cell is missing the link to the enhancer that is in the row.

File: src/documents/SettingsStore.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+A settings store mock that doesn't do persistence. This grade is useful for tests.
+
+<table>
+    <tbody>
+        <tr>
+            <th>Method</th><th>Implementation</th>
+        </tr>
+        <tr>
+            <td><code>get</code></td>
+            <td><code>[fluid.identity](CoreAPI.md#fluid-identity-arg-)</code></td>
```
Comment (jobara) [inline src/documents/SettingsStore.md]: Need to use standard HTML for the link, as the markdown syntax won't work inside of the table.

File: src/documents/Builder.md (hunk 2 above)
Comment (acheetham) [inline src/documents/Builder.md]: @jobara, I'm not quite sure what you're saying here?

Comment (jobara) [inline src/documents/Builder.md]: Sorry, i missed some words there.. basically just asking you to add this link for the enhancer `<a href="to-do/UIEnhancer.md">enhancer</a>`

File: src/documents/tutorial-prefsFramework/CreatingAPrefsEditor.md [hunk 4]
```diff
@@ -767,11 +767,96 @@ radioVolume: {
...
+they're all there.
+How does that happen? Where are they saved to? And how would you change that?
+
+By default, the Preferences Framework automatically saves the preferences to a browser cookie.
+How does that happen?
+* The template has a specific class on the "save" button: `flc-prefsEditor-save`.
+* The Preferences Framework automatically binds a click handler to anything with that class.
+* The click handler ultimately invokes the `set` method on the default
+[settings store](../SettingsStore.md),
+which is a CookieStore.
```
Comment (jobara) [inline src/documents/tutorial-prefsFramework/CreatingAPrefsEditor.md]: Maybe set CookieStore to be a direct link to the cookie store docs?

**Comment by jobara** [comment]
Line 12 of the [UIO Tutorial](https://github.com/fluid-project/infusion-docs/blob/master/src/documents/tutorial-userInterfaceOptions/UserInterfaceOptions.md) could use a link.

**Comment by jobara** [comment]
Is the [Cookie Settings Store](https://github.com/fluid-project/infusion-docs/blob/master/src/documents/to-do/CookieSettingsStore.md) todo page still needed?

File: src/documents/README.md [hunk 5]
```diff
@@ -73,21 +89,31 @@ layout: default
...
-  * [Reorderer](to-do/ReordererAPI.md)
+## Components
+* [Inline Edit](to-do/InlineEditAPI.md)
+* [Pager](to-do/PagerAPI.md)
+* [Progress](to-do/ProgressAPI.md)
+* [Reorderer](to-do/ReordererAPI.md)
     * [List Reorderer](to-do/ListReordererAPI.md)
     * [Grid Reorderer](to-do/GridReordererAPI.md)
     * [Layout Reorderer](to-do/LayoutReordererAPI.md)
     * [Image Reorderer](to-do/ImageReordererAPI.md)
```
Comment (jobara) [inline src/documents/README.md]: There is an actual ImageReordererAPI page now, this should be changed to point at that and the todo one removed.

File: src/documents/README.md [hunk 6]
```diff
@@ -5,66 +5,82 @@ layout: default
...
-  * [Declarative this-ism In IoC](DeclarativeThisismInIoC.md)
-  * [IoC References](IoCReferences.md)
-* Events
-  * [Infusion Event System](InfusionEventSystem.md)
-  * [Event injection and boiling](EventInjectionAndBoiling.md)
-* ChangeApplier
-  * [ChangeApplier](ChangeApplier.md)
-  * [ChangeApplier API](ChangeApplierAPI.md)
-* [Model Relay](ModelRelay.md)
 * [Progressive Enhancement](ProgressiveEnhancement.md)
```
Comment (jobara) [inline src/documents/README.md]: The ProgressiveEnhancement page renders behind the left navigation bar.

Comment (jobara) [inline src/documents/README.md]: Also the link doesn't seem to work when viewing this README file in github.

File: src/documents/Builder.md [hunk 7]
```diff
@@ -144,12 +144,12 @@ The builder object returned by a call to `fluid.prefs.builder()` has the followi
 <tr>
     <td><code>options.assembledPrefsEditorGrade</code></td>
     <td>String</td>
-    <td><a href="ComponentGrades.md">Grade name</a> of the constructed preferences editor; Can be used to instantiate the <a href="PreferencesEditor.md">preferences editor</a>, <a href="to-do/UIEnhancer.md">enhancer</a> and <a href="to-do/SettingsStore.md">settings store</a> using <code><a href="https://github.com/fluid-project/infusion/blob/infusion-1.5/src/framework/core/js/Fluid.js#L930-L944">fluid.invokeGlobalFunction</a></code>.</td>
+    <td><a href="ComponentGrades.md">Grade name</a> of the constructed preferences editor; Can be used to instantiate the <a href="PreferencesEditor.md">preferences editor</a>, <a href="to-do/UIEnhancer.md">enhancer</a> and <a href="SettingsStore.md">settings store</a> using <code><a href="https://github.com/fluid-project/infusion/blob/infusion-1.5/src/framework/core/js/Fluid.js#L930-L944">fluid.invokeGlobalFunction</a></code>.</td>
```
Comment (jobara) [inline src/documents/Builder.md]: I think the "C" in "Can" after the ";" shouldn't be capitalized. Same in the following table row too.

File: src/documents/SettingsStore.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+## fluid.prefs.dataSource ##
+
+A grade used by `fluid.prefs.store`; defines placeholders for `get` and `set` methods.
+
+Method Placeholder  | Description
+------------------- | -----------
+`get`  | For retrieving preferences. Should return the preferences
+`set`  | For saving preferences. Should accept the preferences as an argument.
+
+Note that no particular parameters are prescribed for these methods. It is up to the developer
```
Comment (jobara) [inline src/documents/SettingsStore.md]: Should this use the Infusion-Docs notes styling?

File: src/documents/README.md (hunk 6 above)
Comment (acheetham) [inline src/documents/README.md]: Ah, yes, actually... there is no Progressive Enhancement anymore; replaced by Context Awareness. Removing link.

File: src/documents/SettingsStore.md (hunk 8 above)
Comment (acheetham) [inline src/documents/SettingsStore.md]: I've gone through and found some other notes to style this way, though there were a few that I thought didn't warrant the treatment.

File: src/documents/ChangeApplierAPI.md [hunk 9]
```diff
@@ -185,10 +185,10 @@ that.applier.change("things", "new value");
 
 This example will not log the transition from the initial model state of `undefined` to the console. It will, however, log the value `new value` triggered via the ChangeApplier API.
 
-**NOTE**: The current implementation of the ChangeApplier has a bug ([FLUID-5519](http://issues.fluidproject.org/browse/FLUID-5519)) which will often cause a model listener to be notified
+<div class="infusion-docs-note"><strong>Note:</strong> The current implementation of the ChangeApplier has a bug ([FLUID-5519](http://issues.fluidproject.org/browse/FLUID-5519)) which will often cause a model listener to be notified
```
Comment (jobara) [inline src/documents/ChangeApplierAPI.md]: this `<div>` seems to stripped out from the generated doc.

Comment (jobara) [inline src/documents/ChangeApplierAPI.md]: Never mind, it was cache issue in my browser.

**Comment by jobara** [comment]
merged at aa90f818494e18a8cae9d7cd7ac834aad9254a1a

**Pull request merged**